### DISCUSSION TITLE
Implement PWA on examples/react

### DISCUSSION THREAD

**Discussion by mvperez** [vikejs/vike#882] [Discussion]
The template project I am evaluating to use is `examples/react`. But I would want to add or convert this into PWA. Is possible to do that and how. A reference would be very helpful. Thanks!

**Comment by tr4cks** [comment]
Hey! I'm looking for exactly the same thing. At the moment I've created a `PWA` with `Vite`, `vite-plugin-pwa` as well as `react-router` and I'd also like to know if it's OK to combine it with SSR for SEO.
Do you know if this poses a problem for offline mode with SSR? Are there any tips I should know about?

**Reply by mvperez** [reply]
I turned into NextJs

**Reply by brillout** [reply]
@tr4cks A couple of users are using Vike with vite-plugin-pwa, I believe successfully so. (There were some discussions about it here on Vike's GitHub.) Contribution welcome to create an example (I'll then add it to a dedicated docs page `https://vike.dev/pwa`).

@mvperez What was your reason to go for Next.js instead of Vike?

**Reply by tr4cks** [reply]
Thanks @brillout. I will try to create a `PWA` with `Vike` and I will try to make an example at the same time.
Personally I've used `Next.js` a lot and I like it. The only problem for me is that `PWA` support is a bit limited. The libraries are not well maintained. With Vite, there's a plugin system that has made it possible to create `vite-plugin-pwa` and which is actively maintained.
Maybe I'll come back to you if I have some questions about `Vike` to try to implement a PWA example with SSR rendering.

**Reply by brillout** [reply]
👍 Let me know if you believe there is a blocker on Vike's side. And, yes, an example would be very neat.
